src/js_set.hpp

```diff
@@ -0,0 +1,466 @@
+
+    // set->find will return npos if the element is not found
+    size_t const index = set->find(accessor, value);
+    return_value.set(index != npos);
+}
+
+
+/**
+ * @brief Creates a \ref ResultClass containing a subset of the set's elements
+ * 
+ *  Applies a filter to the elements in the SetClass and returns the elements that match the filter.
+ *  Filters are only supported for sets of objects.
+ *  Will throw `std::runtime_error` if the Set's element type is not objects.
+ * 
+ * @param this_object \ref ObjectType wrapping the SetClass itself
+ * @param args \ref Arguments structure containing the filter that will be applied to the SetClass
+ * @param return_value \ref ReturnValue wrapping a \ref ResultClass containing matching objects to return to the JS context
+ * @exception std::runtime_error Thrown if the \ref SetClass does not contain objects
+ */
+template<typename T>
+void SetClass<T>::filtered(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
+    args.validate_maximum(1);
+
+    auto const set = get_internal<T, SetClass<T>>(ctx, this_object);
+    return_value.set(ResultsClass<T>::create_filtered(ctx, *set, args));
+}
+
+
+/**
+ * @brief Return a textual description of the value element type for the given set
+ * 
+ * @param ctx JS context
+ * @param object /ref ObjectType wrapping the SetClass itself
+ * @param return_value \ref ReturnValue wrapping a static `const char *` descriping the set's element type
+ */
+template<typename T>
+void SetClass<T>::get_type(ContextType ctx, ObjectType object, ReturnValue &return_value) {
+    auto const set = get_internal<T, ListClass<T>>(ctx, object);
+    return_value.set(local_string_for_property_type(set->get_type() & ~realm::PropertyType::Flags));
+}
+
+
+/**
+ * @brief Utility function that validates that elements of a given type is eligible for insertion into the set
+ * 
+ *  Checks whether a given value type is legal for the given SetClass.
+ *  Throws \ref TypeErrorException if the value is not legal.
+ * 
+ * @param ctx JS context
+ * @param set \ref realm::object_store::Set that contains the valid value type
+ * @param value \ref ValueType that is to be checked whether it is valid for the set
+ * @exception TypeErrorException Thrown if `value` is not valid for the set
+ */
+template<typename T>
+void SetClass<T>::validate_value(ContextType ctx, realm::object_store::Set &set, ValueType value) {
+    auto type = set.get_type();
+    StringData object_type;
+    if (type == realm::PropertyType::Object) {
+        object_type = set.get_object_schema().name;
+    }
+    if (!Value::is_valid_for_property_type(ctx, value, type, object_type)) {
+        throw TypeErrorException("Property", object_type ? object_type : local_string_for_property_type(type), Value::to_string(ctx, value));
+    }
+}
+
+template<typename T>
+void SetClass<T>::add_listener(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
```

Comment:
Maybe a comment saying that `args` is validated elsewhere.

Reply:
Done